lms/services/canvas_api.py

```diff
@@ -33,202 +48,351 @@ def get_token(self, authorization_code):
         """
         Get an access token for the current LTI user.
 
-        Posts to the Canvas API to get the access token and returns it.
-
-        :arg authorization_code: The Canvas API OAuth 2.0 authorization code to
+        :param authorization_code: The Canvas API OAuth 2.0 authorization code to
             exchange for an access token
-        :type authorization_code: str
 
-        :raise lms.services.CanvasAPIServerError: if the Canvas API request
-            fails for any reason
+        :raise CanvasAPIServerError: if the Canvas API request fails for any reason
         """
-        response = self._helper.validated_response(
-            self._helper.access_token_request(authorization_code),
-            CanvasAccessTokenResponseSchema,
+        # For documentation of this request see:
+        # https://canvas.instructure.com/doc/api/file.oauth_endpoints.html#post-login-oauth2-token
+
+        parsed_params = self._validated_response(
+            requests.Request(
+                "POST",
+                self._token_url,
+                params={
+                    "grant_type": "authorization_code",
+                    "client_id": self._client_id,
+                    "client_secret": self._client_secret,
+                    "redirect_uri": self._redirect_uri,
+                    "code": authorization_code,
+                    "replace_tokens": True,
+                },
+            ).prepare(),
+            self._CanvasTokenResponseSchema,
```

Comment:
You can now see the request we are going to make. The token calls are a little weird as they can't use the nicer "_make_authenticated_request" interface which creates a request for you.

The schema `_CanvasTokenResponseSchema` mentioned here is directly below these functions so it can be read on the same page as this function.